zap/src/main/java/org/parosproxy/paros/network/HttpSender.java

```diff
@@ -259,6 +262,24 @@ private void checkState() {
         }
     }
 
+    private void updateCookieState() {
+        if (!useCookies) {
+            // discard cookies, but keep credentials
+            HttpState state = new HttpState();
```

Comment:
btw, probably better to not reset the state and just disable/enable cookies, instead we should provide a method to reset the state (so that we can reset it any time from the UI/programmatically).